### DISCUSSION TITLE
Additional Pylontech battery not shown

### DISCUSSION THREAD

**Discussion by dirk-schulz** [hoylabs/OpenDTU-OnBattery#734] [Q&A]
An additional Pylontech battery was added to my setup with openDTU on battery. Pylontech connected as described by vendor. Both batteries communicate together successful. (E.g. start the 2nd battery on power on.)
My assumption was, that openDTU sees 2 Pylontech batteries via CAN Bus and also publish 2 batteries to MQTT.
But only one Pylontech is shown and published to MQTT?!

Has someone experience? Any idea by the developer? Or is it normal behavior, that both batteries act as a single one?
Many thanks in advance for your Feedbacks.

**Comment by Manos1966** [comment]
Hi @dirk-schulz 

Pylontechs act as one.

The only way I can recognise I have two US5000 connected, is that the CANBUS shows 
Charge /Discharge current limit of  160.0  A

![Capture](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/256b0470-4452-4de5-93ea-5f71d354c579)


For this reason I have an additional ESP32 with RS485 connected to the RS485 port as well and get from there additional info #710

**Reply by dirk-schulz** [reply]
@Manos1966 : many thanks!!!
So in your setup the openDTU on battery CAN Bus is only connected to the master.
And a second esp32 is connected via rs485 to both (two cabels)? What are you are using in the 2nd esp32? 
Can you explain a Bit more the config of the 2nd esp32 for RS485? Many thanks in advance

**Reply by Manos1966** [reply]
Happy reading 😄 
#622  (Start from below, work upwards)
skippermeister was kind enough to re-write his fork to handle more than one Pylontech this week (I do not know how this guy finds time to sleep).

- CANbus & RS485 RJ45 cables ONLY on master Pylon
- RS485 cable - Better use a cable with only PIN 7 & 8 connected (all other pins cut/removed)
- Do not connect GND in CANbus & RS485 RJ45 cables
- 2nd ESP32 naked. Just flash an OpenDTUonBattery fork (link inside the diskussion)
- Both OpenDTUs work perfekt in parallel

Welche PYlontech Modelle hast du? Ich bin gespannt ueber manche Werte did dein Pylon BMS ermitteln wird....

**Reply by pelletq** [reply]
Hi,

ich spiele mit dem Gedanken meine US2000C zu erweitern. Nur damit ich das richtig verstehe:
Ich kann die problemlos verbinden wie vorgesehen und meine Kapazität erweitern, der Nachteil ist, dass ich keine gesonderten Daten je Batterie sehe, sondern einfach nur eine "gemischte" mit der kombinierten Kapazität, angezeigt bekomme -korrekt?

 Das fände ich gar nicht so schlimm. Hat das irgendwelche weiteren Nachteile außer der Visualisierung? Oder ist das Problem, dass ich dann eine Dis-Balance oder ähnliches im Laufe der Zeit nicht erkenne (wobei die Pylontechs ja eigentlich dafür gemacht sind und das selber über ihr BMS regeln sollten)?

**Reply by Manos1966** [reply]
@pelletq 
Korrekt. Die OpenDTUonBattery Version ( https://github.com/helgeerbe/OpenDTU-OnBattery ) hat "nur" CANbus Anschluss.
Damit kannst du deine zusaetzliche US2000 problemlos verbinden wie vorgesehen ueber CANbus und deine Kapazität erweitern.
"Nachteil" ist es nicht, eher eine "Einschränkung",  dass du keine gesonderten Daten je Batterie siehst, sondern einfach nur eine "gemischte" mit der kombinierten Kapazität, angezeigt bekommst.

Und sogar diese "Einschraenkung" kannst du einfach beseitigen, indem du eine zusaetzliche ESP32 mit RS485 anschliesst um die extra Daten zu sehen (man kann eine OpenDTUonBattery im CANbus und eine OpenDTU im RS485 gleichzeitig anschliessen.

**Comment by dirk-schulz** [comment]
Many thanks!!!
I will give it a try. But First need to organize the MAX3232.
Wish you a nice weekend and good pv production.

**Reply by Manos1966** [reply]
No MAX3232! 😱 

you need https://www.amazon.de/dp/B0CFF9MWFR
R485 (I told you, start from the bottom 😉 )

And you may want (although expensive 🤮 ) get this, too
https://www.amazon.de/dp/B0CNNPLCZV
unless you know how to take the cable out of a RJ45 plug

By the way, I prefer to have it running on a separate ESP32. 
Better this way, instead of trying to put everything into one ESP32... 👍
